packages/connection/src/Manager.php

```diff
@@ -660,4 +1514,117 @@ public function get_access_token( $user_id = false, $token_key = false, $suppres
+		$this->raw_post_data = $GLOBALS['HTTP_RAW_POST_DATA'];
+		return $methods;
+	}
+
+	/**
+	 * Resets the raw post data parameter for testing purposes.
+	 */
+	public function reset_raw_post_data() {
+		$this->raw_post_data = null;
+	}
+
+	/**
+	 * Registering an additional method.
+	 *
+	 * @param Array $methods an array of available XMLRPC methods.
+	 * @return Array the amended array in case the method is added.
+	 */
+	public function public_xmlrpc_methods( $methods ) {
+		if ( array_key_exists( 'wp.getOptions', $methods ) ) {
+			$methods['wp.getOptions'] = array( $this, 'jetpack_getOptions' );
+		}
+		return $methods;
+	}
+
+	/**
+	 * Handles a getOptions XMLRPC method call.
+	 *
+	 * @param Array $args method call arguments.
+	 * @return an amended XMLRPC server options array.
+	 */
+	public function jetpack_getOptions( $args ) {
+		global $wp_xmlrpc_server;
+
+		$wp_xmlrpc_server->escape( $args );
+
+		$username = $args[1];
+		$password = $args[2];
+
+		$user = $wp_xmlrpc_server->login( $username, $password );
+		if ( ! $user ) {
+			return $wp_xmlrpc_server->error;
+		}
+
+		$options   = array();
+		$user_data = $this->get_connected_user_data();
+		if ( is_array( $user_data ) ) {
+			$options['jetpack_user_id']         = array(
+				'desc'     => __( 'The WP.com user ID of the connected user' ), // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
+				'readonly' => true,
+				'value'    => $user_data['ID'],
+			);
+			$options['jetpack_user_login']      = array(
+				'desc'     => __( 'The WP.com username of the connected user' ), // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
+				'readonly' => true,
+				'value'    => $user_data['login'],
+			);
+			$options['jetpack_user_email']      = array(
+				'desc'     => __( 'The WP.com user email of the connected user' ), // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
+				'readonly' => true,
+				'value'    => $user_data['email'],
+			);
+			$options['jetpack_user_site_count'] = array(
+				'desc'     => __( 'The number of sites of the connected WP.com user' ), // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
+				'readonly' => true,
+				'value'    => $user_data['site_count'],
+			);
+		}
+		$wp_xmlrpc_server->blog_options = array_merge( $wp_xmlrpc_server->blog_options, $options );
+		$args                           = stripslashes_deep( $args );
+		return $wp_xmlrpc_server->wp_getOptions( $args );
+	}
+
+	/**
+	 * Adds Jetpack-specific options to the output of the XMLRPC options method.
+	 *
+	 * @param Array $options standard Core options.
+	 * @return Array amended options.
+	 */
+	public function xmlrpc_options( $options ) {
+		$jetpack_client_id = false;
+		if ( $this->is_active() ) {
+			$jetpack_client_id = \Jetpack_Options::get_option( 'id' );
+		}
+		$options['jetpack_version'] = array(
+			'desc'     => __( 'Jetpack Plugin Version' ), // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
+			'readonly' => true,
+			'value'    => JETPACK__VERSION,
```

Comment:
Got another warning here:

```
[05-Aug-2019 11:02:35 UTC] PHP Warning:  Use of undefined constant JETPACK__VERSION - assumed 'JETPACK__VERSION' (this will throw an Error in a future version of PHP) in /wp-content/plugins/client-example/vendor/automattic/jetpack-connection/src/Manager.php on line 1613
```

Should be resolved if we use the Constants package:

```suggestion
			'value'    => Constants::get_constant( 'JETPACK__VERSION' ),
```